quickwit/quickwit-indexing/src/models/split_attrs.rs

```diff
@@ -120,3 +133,92 @@ pub fn create_split_metadata(
         num_merge_ops: split_attrs.num_merge_ops,
     }
 }
+
+/// reduce the maturity period of a split based on retention policy, so that it doesn't get merged
+/// after it expires.
+fn max_maturity_before_end_of_retention(
```

Comment:
I feel like the code would be clearer and easier to reason with if you defined an order relationship on Maturity.

The relationship would be built upon the idea of "the lower" = "the sooner you reach maturity".
So mature < immature.
immature(x) < immature(y) if x < y.

We could then imply a SplitMaturity based on the retention period. Store that as the "max_maturity" instead of carrying around a RetentionPolicy object, and just compute a `min` here.

Reply:
i agree with implementing Ord for maturity, but I don't think it would remove the need for having the retention policy (or period) passed here: we can't compute a maximum maturity from just the retention period, we also need the split timestamp_range.end

Reply:
You are right.